plugins/com.google.cloud.tools.eclipse.appengine.login.test/src/com/google/cloud/tools/eclipse/appengine/login/ui/AccountSelectorTest.java

```diff
@@ -0,0 +1,184 @@
+    when(account3.getEmail()).thenReturn("some-email-3@example.com");
+    when(account3.getOAuth2Credential()).thenReturn(credential3);
+  }
+
+  @After
+  public void tearDown() {
+    display.dispose();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testConstructor_nullLoginMessage() {
+    new AccountSelector(new Shell(display), loginService, null);
+  }
+
+  @Test
+  public void testComboSetup_noAccount() {
+    when(loginService.getAccounts()).thenReturn(new HashSet<Account>());
+
+    AccountSelector selector = new AccountSelector(shell, loginService, "<select this to login>");
+    assertEquals(-1, selector.combo.getSelectionIndex());
+    assertNull(selector.getSelectedCredential());
+    assertEquals(1, selector.combo.getItemCount());
+    assertEquals("<select this to login>", selector.combo.getItem(0));
+  }
+
+  @Test
+  public void testComboSetup_oneAccount() {
+    when(loginService.getAccounts()).thenReturn(new HashSet<>(Arrays.asList(account1)));
+
+    AccountSelector selector = new AccountSelector(shell, loginService, "<select this to login>");
+    assertEquals(-1, selector.combo.getSelectionIndex());
+    assertNull(selector.getSelectedCredential());
+    assertEquals(2, selector.combo.getItemCount());
+    assertEquals("some-email-1@example.com", selector.combo.getItem(0));
+    assertEquals("<select this to login>", selector.combo.getItem(1));
+  }
+
+  @Test
+  public void testComboSetup_threeAccounts() {
+    when(loginService.getAccounts())
+        .thenReturn(new HashSet<>(Arrays.asList(account1, account2, account3)));
+
+    AccountSelector selector = new AccountSelector(shell, loginService, "<select this to login>");
+    assertEquals(-1, selector.combo.getSelectionIndex());
+    assertNull(selector.getSelectedCredential());
+    assertEquals(4, selector.combo.getItemCount());
+    String allEmails =
+        selector.combo.getItem(0) + selector.combo.getItem(1) + selector.combo.getItem(2);
+    assertTrue(allEmails.contains("some-email-1@example.com"));
+    assertTrue(allEmails.contains("some-email-2@example.com"));
+    assertTrue(allEmails.contains("some-email-3@example.com"));
+    assertEquals("<select this to login>", selector.combo.getItem(3));
+  }
+
+  @Test
+  public void testGetSelectedCredential() {
+    when(loginService.getAccounts())
+        .thenReturn(new HashSet<>(Arrays.asList(account1, account2, account3)));
+    AccountSelector selector = new AccountSelector(shell, loginService, "<select this to login>");
+
+    assertEquals(-1, selector.combo.getSelectionIndex());
+    assertNull(selector.getSelectedCredential());
+    String emailAtIndex1 = selector.combo.getItem(1);
+
+    selector.combo.select(1);
+    selector.logInOnSelect.widgetSelected(mock(SelectionEvent.class));
+
+    assertEquals(1, selector.combo.getSelectionIndex());
+    assertEquals(emailAtIndex1, selector.combo.getItem(1));
+    switch (emailAtIndex1) {
```

Comment:
There shouldn't be a switch like this in a test. It means some of your asserts are not being activated. The test should be deterministic and hit all cases.